src/clj/fluree/db/flake/transact.cljc

```diff
@@ -120,6 +121,9 @@
 (defn stage
   [db fuel-tracker context identity author annotation raw-txn parsed-txn]
   (go-try
+    (when (novelty/max-novelty? db)
+      (throw (ex-info "Maximum novelty exceeded, try again later."
```

Comment:
I think this error message should have better directions. The user would have to instantiate a new db after an index process has completed, so we should mention that.

Reply:
"Max novelty exceeded, wait until indexing is completed."
^ How's that?

Reply:
I went with this:
> "Maximum novelty exceeded, no transactions will be processed until indexing has completed."

Reply:
Only if using the library, correct? I think such a message would confuse someone if they were using fluree/server - so I'd opt to leave that out.

Should it work like this? I think it is odd that a user has a db-t that won't work, but then reloads the same "immutable" db-t to get it to work. Seems like we should be handling that somehow for the user. While I'm not thrilled with it, the new BM25 holds some index state metadata in an atom, and is aware of background updates - replacing/updating the index once complete. Not for this PR - but something I think we should have some thought into.

Reply:
This is a good point re server vs the library. I now agree that we should leave the reloading the db part out. If someone is using the db library, they probably would have a better understanding about how indexing works anyway.

I think there are definitely warts with this approach forcing the user to update their own db after the fact, but there's a lot of complexity involved with keeping a handle of all dbs that have been instantiated and having them update themselves as well. I agree that we should probably handle this for the user eventually though.

Reply:
I'm glad you mentioned that Brian, because what I came up with is "... Obtain a new Transactable record with a reference to the new index." And that seemed way too inside-baseball for a regular user.